tests/periph_uart_mode/main.c

```diff
@@ -52,6 +54,25 @@
 /* Stores each mode string for printing at the end of the test */
 static char mode_strings[TOTAL_OPTIONS][MODE_STR_LEN];
 
+static void _delay(void)
+{
+    if (IS_USED(MODULE_XTIMER)) {
+        xtimer_usleep(DELAY_US);
+    }
+    else {
+        /*
+         * As fallback for freshly ported boards with no timer drivers written
+         * yet, we just use the CPU to delay execution and assume that roughly
+         * 20 CPU cycles are spend per loop iteration.
+         *
+         * Note that the volatile qualifier disables compiler optimizations for
+         * all accesses to the counter variable. Without volatile, modern
+         * compilers would detect that the loop is only wasting CPU cycles and
+         * optimize it out - but here the wasting of CPU cycles is desired.
+         */
+        for (volatile uint32_t i = 0; i < CLOCK_CORECLOCK / 20; i++) { }
```

Comment:
we could really do with a `busy_wait()` helper function somewhere

Reply:
Maybe even one that prefers to call into ztimer / xtimer instead, if already used.

Reply:
why not both?
A `busy_wait()` function that will never sleep can be handy, but so can a `wait_usec()` function that can be used for *very* portable code.